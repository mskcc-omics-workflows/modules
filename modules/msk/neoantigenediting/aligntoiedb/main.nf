process NEOANTIGENEDITING_ALIGNTOIEDB {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigenediting:1.2':
        'docker.io/mskcc/neoantigenediting:1.2' }"

    input:
    tuple val(meta),  path(patient_data)
    path(iedb_fasta)

    output:
    tuple val(meta), path("iedb_alignments_*.txt")             , emit: iedb_alignment
    path "versions.yml"                                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    python3 /usr/bin/align_neoantigens_to_IEDB.py \\
        --fasta ${iedb_fasta} \\
        --input ${patient_data}



    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        neoantigenEditing: \$NEOANTIGEN_EDITING_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch iedb_alignments_example.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        neoantigenEditing: \$NEOANTIGEN_EDITING_TAG
    END_VERSIONS
    """
}
