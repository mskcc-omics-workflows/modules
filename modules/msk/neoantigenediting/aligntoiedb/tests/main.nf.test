nextflow_process {

    name "Test Process NEOANTIGENEDITING_ALIGNTOIEDB"
    script "../main.nf"
    process "NEOANTIGENEDITING_ALIGNTOIEDB"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "neoantigenediting"
    tag "neoantigenediting/aligntoiedb"

    test("neoantigenediting_aligntoiedb - json") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true)
                    ]
                input[1] = file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.iedb_alignment[0][1]).name).match() }

            )
        }

    }

    test("neoantigenediting_aligntoiedb - json - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('patient_data')
                    ]
                input[1] = file('iedb_fasta')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }


}
