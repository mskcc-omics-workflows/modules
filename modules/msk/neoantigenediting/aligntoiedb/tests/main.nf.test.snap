{
    "neoantigenediting_aligntoiedb - json - stub": {
        "content": [
            [
                "versions.yml:md5,ee206e836fcb1b31809f63f2ffa49519"
            ],
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "iedb_alignments_example.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee206e836fcb1b31809f63f2ffa49519"
                ],
                "iedb_alignment": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "iedb_alignments_example.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee206e836fcb1b31809f63f2ffa49519"
                ]
            }
        ],
        "timestamp": "2024-06-19T14:54:25.768151"
    },
    "neoantigenediting_aligntoiedb - json": {
        "content": [
            [
                "versions.yml:md5,ee206e836fcb1b31809f63f2ffa49519"
            ],
            "iedb_alignments_3-OLTS.txt"
        ],
        "timestamp": "2024-06-19T14:54:19.606216"
    }
}